src/containers/SimulationContainer/index.js

```diff
@@ -55,89 +71,70 @@ class SimulationContainer extends Component {
         // I'm still alive, keep polling.
       } else {
         // I finished processing
-        clearInterval(this.timer);
+        clearInterval(timer.current);
         if (status === "processed") {
-          this.fetchSimulationResult(resultUrl);
+          fetchSimulationResult(resultUrl);
         } else {
-          this.showError({ message: lastError });
+          showError({ message: lastError });
         }
       }
     } else {
-      clearInterval(this.timer);
-      this.showError(response.payload);
+      clearInterval(timer.current);
+      showError(response.payload);
     }
-  }
+  };
 
-  async fetchSimulationResult(url) {
-    const response = await this.fetchJSON(url);
+  const fetchSimulationResult = async (url) => {
+    const response = await fetchJSON(url);
     if (response.success) {
-      this.setState({
-        loading: false,
-        jsonPayload: response.payload,
-      });
+      setLoading(false);
+      setJsonPayload(response.payload);
     } else {
-      this.showError(response.payload);
+      showError(response.payload);
     }
-  }
-
-  showError(error) {
-    this.setState({
-      loading: false,
-      errorMessage: error.message,
-    });
-  }
+  };
 
-  render() {
-    const { loading, errorMessage, jsonPayload } = this.state;
-    const isLoaded = !loading;
-    const hasError = !!errorMessage;
-    const isSuccess = isLoaded && !hasError;
-    const { classes, history, set, location } = this.props;
-    const routeMatch = matchPath(location.pathname, {
-      path: "/simulations/:simulationId",
-      exact: true,
-      strict: false,
-    });
-    const simulationId = routeMatch && (routeMatch.params || {}).simulationId;
-    const open = !!simulationId;
+  const showError = error => {
+    setLoading(false);
+    setErrorMessage(error.message);
+  };
 
-    return (
-      <Dialog
-        fullScreen
-        open={open}
-        className={classes.root}
-        onClose={() => history.push("/simulations")}
-        TransitionComponent={Transition}
-      >
-        <TopBar fullscreen backLinkPath="/simulations">
-          <Typography
-            variant="h6"
-            color="inherit"
-            className={classes.appBarHeader}
-          >
-            {set.name}
-          </Typography>
-          <FiltersToggleBtn variant="info" className={classes.filtersBtn} />
-        </TopBar>
-        <Fade in={loading} unmountOnExit>
-          <LinearProgress variant="query" />
-        </Fade>
-        <Snackbar
-          anchorOrigin={{ vertical: "top", horizontal: "right" }}
-          open={isLoaded && hasError}
-          autoHideDuration={6000}
-          message={<span id="message-id">Error: {errorMessage}</span>}
-        />
-        {isSuccess && (
-          <SimulationList key="list" simulations={jsonPayload.invoices} />
-        )}
-      </Dialog>
-    );
-  }
-}
+  return (
+    <Dialog
```

Comment:
This should move to `/components/Simulation/index.js` to keep containers html free.